test/mocha/10-canonize-hash-api.js

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2017-2021 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const {util: {clone}} = require('bedrock');
+const brLedgerNode = require('bedrock-ledger-node');
+const mockData = require('./mock.data');
+
+const rdfCanonizeAndHash = brLedgerNode.consensus._rdfCanonizeAndHash;
+
+describe('Canonize And Hash API', function() {
+  describe('WebLedgerConfiguration', function() {
+    it('should canonize & hash', async function() {
+      const data = clone(mockData.ledgerConfiguration);
+      const result = await rdfCanonizeAndHash(data);
+      should.exist(result);
+      result.should.be.an('object');
+      result.should.have.property('canonizedBytes');
+      result.should.have.property('hash');
+      result.should.eql(mockData.canonize.ledgerConfiguration);
+    });
+    it('should have a unique hash', async function() {
+      const data = clone(mockData.ledgerConfiguration);
+      data.ledger = 'foo';
+      const result = await rdfCanonizeAndHash(data);
+      should.exist(result);
+      result.should.be.an('object');
+      result.should.have.property('canonizedBytes');
+      result.should.have.property('hash');
+      result.should.not.eql(mockData.canonize.ledgerConfiguration);
+    });
+    it('should drop terms not in the "@context"', async function() {
```

Comment:
Should the behavior here to be throw instead of silently dropping?

cc: @dlongley @davidlehn

Reply:
Just to confirm in both `jsonld` 2 and 5 it is not throwing, and nothing in the CHANGELOG mentions throwing on terms that are not in the `@context`: https://github.com/digitalbazaar/jsonld.js/blob/master/CHANGELOG.md

The spec implies that terms not in the `@context` should be ignored:

https://w3c.github.io/json-ld-syntax/#example-term-expansion-from-context-definition

But I might be reading that out of context (pun not intended) as that concerns expansion.

Reply:
I can't speak to what the behavior should be in this use case, but as far as implementation, if you wanted to throw, it's an implementation issue.  Run expand() first, with an expansionMap that throws when needed, and pass result to canonize with skipExpansion.  See jsonld-signatures lib/expandMap.js.

Reply:
This is testing the behavior the code has now.  So it's probably fine, and just update it if the behavior changes.

Reply:
Okay, I did not know whether we wanted to apply the same tactic in jsigs throughout our entire usage of jsonld. I'm fine with continuing to test the same behavior and come back to the throw vs. silently dropping in the future. Just so that my opinion is recorded, my preference would be to throw.